pkg/sync/service.go

```diff
@@ -233,10 +235,16 @@ func (s *Service) DownloadToDst(writer http.ResponseWriter, request *http.Reques
 	}
 	defer src.Close()
 
-	writer.Header().Set("Content-Disposition", "attachment; filename="+sf.uuid)
+	gzipWriter := gzip.NewWriter(writer)
+	defer gzipWriter.Close()
+
+	// e.g. filePath=/data/backing-images/parrot-6846a0b2/backing
+	filePathSlices := strings.Split(sf.filePath, "/")
+	backingImageNameUUID := filePathSlices[len(filePathSlices)-2]
+
+	writer.Header().Set("Content-Disposition", fmt.Sprintf("attachment; filename=%s.gz", strings.Split(backingImageNameUUID, "-")[0]))
 	writer.Header().Set("Content-Type", "application/octet-stream")
-	writer.Header().Set("Content-Length", strconv.FormatInt(sf.size, 10))
```

Comment:
Removing content-length is because we can't determine the streaming download, right? This header can be optional.

Reply:
Yes and also because it is now compressed so we don't know the exact size.
user should always use checksum to check it